kolibri/core/device/api.py

```diff
@@ -164,3 +168,37 @@ def patch(self, request):
         settings.name = request.data["name"]
         settings.save()
         return Response({"name": settings.name})
+
+
+class UserSyncStatusViewSet(ReadOnlyValuesViewset):
+    values = (
+        "id",
+        "queued",
+        "sync_session",
+        "user",
+    )
+
+    def get_queryset(self):
+        queryset = UserSyncStatus.objects.all()
+        classroom_id = self.request.query_params.get("classroom_id")
+        if classroom_id is not None:
+            classroom = Classroom.objects.get(pk=classroom_id)
+            classroom_users = FacilityUser.objects.filter(
+                memberships__collection=classroom
+            )
+            queryset = queryset.filter(user__in=classroom_users)
+        return queryset
+
+
+class ClassListSyncStatusViewSet(ReadOnlyValuesViewset):
+    values = (
```

Comment:
As the only difference between these two Viewsets is filtering that is done, I'd recommend consolidating these and using a FilterSet to return different sets of data.

```
class SyncStatusFilter(FilterSet):

    member_of = ModelChoiceFilter(
        method="filter_member_of", queryset=Collection.objects.all()
    )

    def filter_member_of(self, queryset, name, value):
        return queryset.filter(Q(user__memberships__collection=value) | Q(user__facility=value))

    class Meta:
        model = UserSyncStatus
        fields = ["user", "member_of"]
```

Can then add to the viewset as:
```
filter_backends = (KolibriAuthPermissionsFilter, DjangoFilterBackend)
filter_class = SyncStatusFilter
```

We don't actually need `KolibriAuthPermissionsFilter` on there, but we should also add the permission based filtering to prevent unauthorized access.